frontend/src/features/mapSideBar/SidebarContents/LandForm.tsx

```diff
@@ -252,6 +267,8 @@ interface ILandForm {
   handlePidChange: (pid: string) => void;
   /** help with formatting of the pin */
   handlePinChange: (pin: string) => void;
+  /** Function that searches for a parcel matching a pid within the API */
+  findMatchingPid: (pid: string, nameSpace?: string | undefined) => Promise<IParcel | undefined>;
```

Comment:
I can't remember if I've asked this before, what's a promise?

Reply:
Thats too big of a question to answer here. In summary its an es6 feature that allows you to handle async actions. There are a lot of tutorials out there on what a promise is, here is one: https://codeburst.io/a-simple-guide-to-es6-promises-d71bacd2e13a

Reply:
@agalbrai I am no expert by any means but normally with a promise it starts out in as `pending` and then the promise is either `fulfilled` or `rejected`. Then you can have your error handling when it is rejected, and whatever actions you would like to take next when it is fulfilled. You'll see some `.then()'s` scattered throughout our app and whatever is inside of that is what happens on fulfillment / `.catch()` to deal with the rejected promises

Reply:
Also here's a handy image
![image](https://user-images.githubusercontent.com/15724124/107818415-6aa32f00-6d2c-11eb-9c4e-c6f0eb825aab.png)

Reply:
thanks so much!